Update sql-database-aad-authentication-configure.md

SvenAelterman opened the pull request:
Clarification on initial Azure AD managed domain
Change reference to "Windows group" to "Azure AD" group
Clarification on federated user account as external user in a managed domain

azurecla commented:
Hi __@devinrader__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, AZPRBOT;

SvenAelterman committed:
Update sql-database-aad-authentication-configure.md

Clarification on initial Azure AD managed domain
Change reference to "Windows group" to "Azure AD" group
Clarification on federated user account as external user in a managed domain

BYHAM reviewed (approved):
Yes. Good changes. Thank you.
